src/middlewares/errorMiddleware.js

```diff
@@ -0,0 +1,22 @@
+const ApiError = require('../exeptions/api.error.js');
+
+const errorMiddleware = (error, req, res, next) => {
+  if (error instanceof ApiError) {
+    res.status(error.status).send({
+      message: error.message,
+      errors: error.errors,
+    });
+  }
+
+  res.statusCode = 500;
+
+  res.send({
+    message: 'Server error',
+  });
+
+  next();
```

Comment:
The `next()` function should not be called after sending a response, as it can lead to multiple responses being sent. Consider restructuring the code to ensure that `next()` is only called if no response is sent.